### PR TITLE
[FxImporter] refactor canonicalize using table driven

### PR DESCRIPTION
**Pull request by penguin-wwy**

### CODE REVIEW

**Comment by penguin-wwy** [comment]
@stellaraccident please review this

**Comment by penguin-wwy** [comment]
https://github.com/llvm/torch-mlir/pull/3387#pullrequestreview-2084268493
@stellaraccident

File: python/torch_mlir/extras/fx_importer.py [hunk 1]
```diff
@@ -2401,3 +2360,98 @@ def _ref_finalizer(self, ref_id: int):
...
     "torch.aten.floor_divide": "torch.aten.floor_divide.Scalar",
 }
+
+
+NODE_CANONICALIZE: Dict[TorchOpOverload, Callable] = {}
+
+
+def register_canonicalize(op: TorchOpOverload):
+    def wrapper(func):
+        global NODE_CANONICALIZE
```
Comment (stellaraccident) [inline python/torch_mlir/extras/fx_importer.py:2370]: You don't need this of just reading. Having it here is a strong signal to the reader that you intend to update the global, so better to not leave it out.

**Review by stellaraccident — APPROVED** [review]
Sorry I missed this

File: python/torch_mlir/extras/fx_importer.py (hunk 1 above)
Comment (penguin-wwy) [reply python/torch_mlir/extras/fx_importer.py]: done

**Pull request merged**